src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx

```diff
@@ -1103,13 +1107,148 @@ func socialSignInWithWebUI() -> AnyCancellable {
+
+If the desired first factor is known before the sign in flow is initiated it can be passed to the initial sign in call:
+
+```ts
+type AuthFactorType = 
+    | 'WEB_AUTHN' 
+    | 'EMAIL_OTP' 
+    | 'SMS_OTP' 
+    | 'PASSWORD' 
+    | 'PASSWORD_SRP';
+
+type SignInOptions = AuthServiceOptions & {
+    authFlowType?: AuthFlowType;
+    clientMetadata?: ClientMetadata;
+    preferredChallenge?: AuthFactorType;
+};
+```
+
+```ts
+// PASSWORD_SRP / PASSWORD
+// sign in with preferred challenge as password
+// note password must be provided in same step
+const { nextStep } = await signIn({
+    username: "hello@mycompany.com",
+    password: "hunter2",
+    options: {
+        authFlowType: "USER_AUTH",
+        preferredChallenge: "PASSWORD_SRP" // or "PASSWORD"
+    },
+});
+
+// nextStep.signInStep === 'DONE'
+handleNextSignInStep(nextStep);
+
+// WEB_AUTHN
+// sign in with preferred challenge as web authn
+// no user input required at this step 
+const { nextStep } = await signIn({
+    username: "hello@mycompany.com",
+    options: {
+        authFlowType: "USER_AUTH",
+        preferredChallenge: "WEB_AUTHN" 
+    },
+});
+
+// nextStep.signInStep === 'DONE'
+handleNextSignInStep(nextStep);
+
+// EMAIL_OTP
+// sign in with preferred challenge as email otp
+// no user input required at this step 
+const { nextStep } = await signIn({
+    username: "hello@mycompany.com",
+    options: {
+        authFlowType: "USER_AUTH",
+        preferredChallenge: "EMAIL_OTP" 
+    },
+});
+
+// nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_EMAIL_CODE'
+handleNextSignInStep(nextStep);
+
+// SMS_OTP
+// sign in with preferred challenge as sms otp
+// no user input required at this step 
+const { nextStep } = await signIn({
+    username: "hello@mycompany.com",
+    options: {
+        authFlowType: "USER_AUTH",
+        preferredChallenge: "SMS_OTP" 
+    },
+});
+
+// nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_SMS_CODE'
+handleNextSignInStep(nextStep);
+```
+
+Note that if the preferred auth factor is not available, the preference will be ignored and the flow will continue to first factor discovery.
+
+### Passwordless Sign In with First Factor Discovery
+
+When multiple first factor options are available and none are set as preferred in InitiateAuth API call (or preferred challenge is not available), Amplify JS continues to a new sign in step `CONTINUE_SIGN_IN_WITH_FIRST_FACTOR_SELECTION`:
+
+```ts
+interface ContinueSignInWithFirstFactorSelection {
+    signInStep: 'CONTINUE_SIGN_IN_WITH_FIRST_FACTOR_SELECTION';
+    availableChallenges?: ChallengeName[];
+}
+```
+
+```ts
+const { nextStep } = await signIn({
+    username: "hello@mycompany.com",
+    options: {
+        authFlowType: "USER_AUTH",
+    }
+});
+
+// nextStep.signInStep === 'CONTINUE_SIGN_IN_WITH_FIRST_FACTOR_SELECTION'
+handleNextSignInStep(nextStep);
```

Comment:
I think we should avoid including internal types in the documentation; we should show an example of how to use the availableChallenges instead.